core/io/json.cpp

```diff
@@ -118,7 +118,7 @@ String JSON::_stringify(const Variant &p_var, const String &p_indent, int p_cur_
 			p_markers.insert(d.id());
 
 			List<Variant> keys;
-			d.get_key_list(&keys);
+			d.get_key_list(&keys, true);
```

Comment:
why not use `keys.sort_custom<StringLikeVariantOrder>()` instead of this interface change?

Reply:
Look above, this is an alternative solution I'm proposing 🙃